3723/1/repo.rb

```diff
@@ -0,0 +1,27 @@
+require 'json'
+require 'yaml'
+require 'faraday'
+
+class Repo
+  attr_reader :list
+
+  def initialize
+    @list = []
+  end
+
+  def write_gem(info)
+    response = Faraday.get "https://api.github.com/search/repositories?q=#{info}"
+    data = JSON.parse(response.body)
+    repo = data['items'].first['full_name']
+    @list << "https://github.com/#{repo}"
+  end
+
+  def take_repo(doc)
```

Comment:
And if you place `write_gem` as private, `take_repo` will be the only one public method in the class. It's the best approach to achieve Single Responsibility Principle (the first one of SOLID). I'd recommend to follow the principle from the very beginning.

Additionally, I'd vote for another name for the class, because it doesn't represent a repo. It represent an action of taking information from the repo. So it might have name like `RepoInfoLoader` which is an action and the only one public method might be renamed into `call` (which is a very common convention)

Reply:
Done